core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -3564,6 +3564,24 @@ private SqlDialect nonOrdinalDialect() {
         .withDoris().ok(expectedStarRocks);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-7069">[CALCITE-7069]
+   * Invalid unparse for INT UNSIGNED and BIGINT UNSIGNED in MysqlSqlDialect</a>. */
+  @Test void testCastToUnsignedInMySQL() {
+    final String query1 = "select cast(\"product_id\" as bigint unsigned) from \"product\"";
+    // MySQL does not allow cast to BIGINT UNSIGNED; instead cast to UNSIGNED.
+    final String expectedMysql1 = "SELECT CAST(`product_id` AS UNSIGNED)\n"
+        + "FROM `foodmart`.`product`";
+    final String query2 = "select cast(\"product_id\" as integer unsigned) from \"product\"";
+    // MySQL does not allow cast to INTEGER UNSIGNED; instead cast to UNSIGNED.
```

Comment:
Here, the specific types have been removed. There might have been some use cases that required throwing exceptions, but they don't exist anymore now.

Reply:
> Here, the specific types have been removed. There might have been some use cases that required throwing exceptions, but they don't exist anymore now.

SELECT cast(18446744073709551615 as unsigned integer) is equivalent to SELECT cast(18446744073709551615 as unsigned) in MySQL.

Reply:
There should have been an exception thrown for crossing the boundary, but MySQL did not.

Reply:
only if the result is the same. Maybe cast to UNSIGNED actually truncates the result - then the result is not the same.